ci/build_wheel.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION
+
+# install CUDAToolKit version 12.2
+apt-get update
+apt-get -y install cuda-toolkit-12-2
+
+rapids-logger "Check conda environment"
+
+conda list
+
+rapids-logger "Building wheel"
+
+pip wheel .
+
+rapids-logger "Installing wheel"
+
+pip install pynvjitlink-0.1.0-cp310-cp310-manylinux_2_35_x86_64.whl
```

Comment:
I just noticed that we're going to need a Python version matrix. This wheel only support Python 3.10. We need to ship pynvjitlink on Python 3.9, 3.10 for RAPIDS, and also 3.11 and 3.12 if we want to make external consumers from Numba happy. We don't have Python 3.11 / 3.12 Docker images yet.

Reply:
@gmarkall Can you comment on the expected Python version support for this package?

Reply:
I was hoping to follow [SPEC 0](https://scientific-python.org/specs/spec-0000/#support-window), so at the present time 3.10-3.12 support would be my aim. If 3.9 support is also needed for RAPIDS then that seems OK. If we can't initially provide wheels for 3.11 and 3.12 then I think that's also OK.

My reasoning is that for non-RAPIDS use cases, it will always be possible for users to build pynvjitlink themselves (unlike with cubinlinker), and we're covering the RAPIDS deployment scenarios in supporting 3.9 and 3.10 from the start. If there's no short-term route to subsequently adding 3.11 and 3.12, I'd like to look into how to get to supporting them eventually.

Reply:
It looks like there aren't `ci-conda` centos images spanning the range of python versions we need. Should we be installing the right version of python ourselves within the container or is there something else I should be doing instead?

Reply:
You can add newer Python CI images, or just use Python versions from conda. Perhaps it will suffice to have a single build job that iterates over all the Python versions, since the builds should be fast.

Reply:
> What do you mean by "add newer python CI images"?

Today, our CI images only cover Python 3.9 and 3.10. https://github.com/rapidsai/ci-imgs/blob/bf7d540855e7e94695a54e40080c3acb8681764a/matrix.yaml#L8-L10

Reply:
> Perhaps it will suffice to have a single build job that iterates over all the Python versions, since the builds should be fast.

I had in mind using a conda variant build like in ucxx, where the `conda_build_config.yaml` has a list of Python versions and `conda build` knows to build all of them.

https://github.com/rapidsai/ucxx/blob/4464ca942d5ac71aabe4492e2a6d59da453f06dc/conda/recipes/ucxx/conda_build_config.yaml#L19-L21